File: tests/fixtures/fake_spring_app/src/main/java/com/example/demo/UserController.java
package com.example.demo;

import org.springframework.web.bind.annotation.*;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/users")
public class UserController {

    @GetMapping
    public List<User> getAllUsers(@RequestParam(required = false) String role) {
        return List.of();
    }

    @GetMapping("/{id}")
    public User getUserById(@PathVariable Long id) {
        return new User();
    }

    @PostMapping
    public User createUser(@RequestBody Map<String, Object> userData) {
        return new User();
    }

    @PutMapping("/{id}")
    public User updateUser(@PathVariable Long id, @RequestBody Map<String, Object> userData) {
        return new User();
    }

    @DeleteMapping("/{id}")
    public void deleteUser(@PathVariable Long id) {
    }

    @GetMapping("/search")
    public List<User> searchUsers(
            @RequestParam String query,
            @RequestParam(required = false, defaultValue = "10") Integer limit) {
        return List.of();
    }
}

File: tests/fixtures/fake_kotlin_spring_app/src/main/kotlin/com/example/api/UserController.kt
package com.example.api

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/users")
class UserController {

    @GetMapping
    fun listUsers(@RequestParam(required = false) role: String?): List<User> {
        return emptyList()
    }

    @GetMapping("/{id}")
    fun getUser(@PathVariable id: Long): User {
        return User()
    }

    @PostMapping
    fun createUser(@RequestBody body: UserRequest): User {
        return User()
    }

    @PutMapping("/{id}")
    fun updateUser(@PathVariable id: Long, @RequestParam name: String): User {
        return User()
    }

    @DeleteMapping("/{id}")
    fun deleteUser(@PathVariable id: Long) {
    }
}
